bin/cron/csf_workshop_attendance_to_gdrive

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env ruby
+require_relative 'only_one'
+abort 'Script already running' unless only_one_running?(__FILE__)
+
+require_relative '../../dashboard/config/environment'
+require 'cdo/google/sheet'
+
+#
+# Uses a Google Cloud service account to write enrollment data for CSF workshops into
+# a spreadsheet in Google Drive (with permissions locked down to our organization).
+#
+
+EARLIEST_WORKSHOP_END_DATE = Date.new(2020, 6, 1)
+LATEST_WORKSHOP_START_DATE = Date.today
+
+def get_rows
+  # Seed results array with a header row
+  results = [
+    [
+      "Organizer Name",
+      "Organizer Email",
+      "Regional Partner Name",
+      "Workshop Dates",
+      "Funded",
+      "Attendance Url",
+      "Num Facilitators",
+      "Course",
+      "Subject",
+      "Is this a Virtual Workshop?",
+      "Total Hours",
+      "Num Registered",
+      "Num Scholarship Teachers Attending Day 1",
+      "Attendance Day 1"
+    ]
+  ]
+
+  Pd::Workshop.
+    where(subject: Pd::SharedWorkshopConstants::SUBJECT_CSF_101).
+    scheduled_end_on_or_after(EARLIEST_WORKSHOP_END_DATE).
+    scheduled_start_on_or_before(LATEST_WORKSHOP_START_DATE).find_each do |workshop|
+      results << [
+        workshop.organizer&.name,
+        workshop.organizer&.email,
+        workshop.regional_partner&.name,
+        workshop.sessions.map(&:formatted_date_with_start_and_end_times).join("\n"),
+        workshop.funded ? "Yes" : "No",
+        "https://studio.code.org/pd/workshop_dashboard/workshops/#{workshop.id}",
+        workshop.facilitators.count,
+        workshop.course,
+        workshop.subject,
+        workshop.virtual? ? "Yes" : "No",
+        workshop.effective_num_hours,
+        workshop.enrollments.count,
+        scholarship_teacher_attendance(workshop),
+        workshop.sessions.first.attendances.count
+      ]
+    end
+
+  results
+end
+
+def scholarship_teacher_attendance(workshop)
+  num_teachers = 0
+  workshop.sessions.first.attendances.each do |attendance|
+    if Pd::ScholarshipInfo.where(pd_enrollment_id: attendance.pd_enrollment_id)
```

Comment:
Isn't "no" a valid scholarship status? Would this mark those teachers as scholarship recipients?

Reply:
We also have this pretty similar method in the workshop model that I think I added for something similar last year, I wonder if it's worth generalizing / making sure that the way "scholarship attendance" is being calculated here is consistent with how we're doing the same thing elsewhere?

https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/models/pd/workshop.rb#L687

Reply:
oh excellent point - I totally forgot "No" is a status! And thanks for pointing me to this method! This export was initially getting attendance for each session, but now that I don't need to do that, I can reuse the method from the model. Thanks!